From 9027a40f2039375f7f4eae519e388999c4530a4f Mon Sep 17 00:00:00 2001
From: Matt Jeanneret <mjeanneret@microsoft.com>
Date: Sat, 28 Jan 2023 17:34:53 -0500
Subject: [PATCH] Allocate 2 cpu for the emulator thread instead of one

The ensures SMT alignment when cpu-manager policy
full-pcpus-only is used. Preventing a dedicated cpu
from sharing time with the emulator thread/virt-launcher
processes.
---
 pkg/virt-controller/services/renderresources.go | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/pkg/virt-controller/services/renderresources.go b/pkg/virt-controller/services/renderresources.go
index 49f3d4a18..84dadd2db 100644
--- a/pkg/virt-controller/services/renderresources.go
+++ b/pkg/virt-controller/services/renderresources.go
@@ -194,9 +194,11 @@ func WithCPUPinning(cpu *v1.CPU) ResourceRendererOption {
 			}
 		}
 
-		// allocate 1 more pcpu if IsolateEmulatorThread request
+		// allocate 2 more pcpu if IsolateEmulatorThread request
+		// the extra cpu ensures hyperthread SMT cpu alignment when cpu-manager
+		// policy full-pcpus-only is used.
 		if cpu.IsolateEmulatorThread {
-			emulatorThreadCPU := resource.NewQuantity(1, resource.BinarySI)
+			emulatorThreadCPU := resource.NewQuantity(2, resource.BinarySI)
 			limits := renderer.calculatedLimits[k8sv1.ResourceCPU]
 			limits.Add(*emulatorThreadCPU)
 			renderer.calculatedLimits[k8sv1.ResourceCPU] = limits
-- 
2.25.1

